packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -1306,6 +1307,17 @@ List<DiagnosticsNode> _debugCompareFloats(
 /// than zero, then it should override [childCrossAxisPosition]. For example
 /// [RenderSliverGrid] overrides this method.
 abstract class RenderSliver extends RenderObject {
+  /// Whether this sliver should be included in the semantics tree.
+  ///
+  /// This value is used by [RenderViewportBase.visitChildrenForSemantics]
+  /// to ensure a sliver is included in the semantics tree regardless of its geometry.
+  ///
+  /// [RenderSliverEnsureSemantics] overrides this value to `true` to ensure
+  /// its sliver child is included in the semantics tree.
```

Comment:
somewhere in the doc should explain the following

1. the difference between this and off screen rendersliver that was laid out due to cache extent
2. A typical example why someone may want to override this getter to return true.